src/Response/ResponseExtensions.cs

```diff
@@ -58,5 +63,118 @@ public static class ResponseExtensions
+        /// </summary>
+        /// <param name="response">Current <see cref="HttpResponse"/></param>
+        /// <param name="stream">The <see cref="Stream"/> to copy from</param>
+        /// <param name="contentType">The content type for the response</param>
+        /// <param name="contentDisposition">The content disposition to allow file downloads</param>
+        /// <returns><<see cref="Task"/>/returns>
+        public static async Task FromStream(this HttpResponse response, Stream source, string contentType, ContentDisposition contentDisposition = null)
+        {
+            long rangeStart;
+            long rangeEnd;
+            var contentLength = source.Length;
+
+            response.Headers["AcceptRanges"] = "bytes";
+            response.ContentType = contentType;
+
+            if (contentDisposition != null)
+            {
+                response.Headers["Content-Disposition"] = contentDisposition.ToString();
+            }
+
+            // rangeHeader should be of the format "bytes=0-" or "bytes=0-12345" or "bytes=123-456"
+            var rangeHeader = response.HttpContext.Request.Headers["Range"].FirstOrDefault();
+
+            if (!string.IsNullOrEmpty(rangeHeader) && rangeHeader.Contains("="))
+            {
+                var rangeParts = rangeHeader.SplitOnFirst("=")[1].SplitOnFirst("-");
+                rangeStart = long.Parse(rangeParts[0]);
+                rangeEnd = rangeParts.Length == 2 && !string.IsNullOrEmpty(rangeParts[1])
+                    ? int.Parse(rangeParts[1]) // the client requested a chunk
+                    : contentLength - 1;
+
+                if (rangeStart < 0 || rangeEnd > contentLength - 1)
+                {
+                    response.StatusCode = (int)HttpStatusCode.RequestedRangeNotSatisfiable;
+                    await response.WriteAsync(string.Empty);
+                }
+                else
+                {
+                    response.Headers["ContentRange"] = $"bytes {rangeStart}-{rangeEnd}/{contentLength}";
+                    response.StatusCode = (int)HttpStatusCode.PartialContent;
+                    await GetResponseBodyDelegate(rangeStart, rangeEnd, contentLength, source, response.Body, response.HttpContext.RequestAborted);
+                }
+            }
+            else
+            {
+                rangeStart = 0;
+                rangeEnd = contentLength - 1;
+                await GetResponseBodyDelegate(rangeStart, rangeEnd, contentLength, source, response.Body, response.HttpContext.RequestAborted);
+            }
+        }
+
+        private static async Task GetResponseBodyDelegate(long rangeStart, long rangeEnd, long contentLength, Stream source, Stream destination, CancellationToken cancellationToken)
+        {
+            if (rangeStart == 0 && rangeEnd == contentLength - 1)
+            {
+                await StreamCopyOperation.CopyToAsync(source, destination, new long?(), 65536, cancellationToken);
```

Comment:
Maybe make the `rangeStart` and `rangeEnd` nullable to basically disable the splitting?

Also, maybe return here, so you don’t need to indent the whole rest of the method?